src/escpos/escpos.py

```diff
@@ -578,9 +578,11 @@ def cut(self, mode=''):
         # Fix the size between last line and cut
         # TODO: handle this with a line feed
         self._raw(b"\n\n\n\n\n\n")
-        if mode.upper() == "PART":
+
+        if mode.upper() == "PART" and self.profile.supports('paperPartCut'):
             self._raw(PAPER_PART_CUT)
-        else:  # DEFAULT MODE: FULL CUT
+        elif mode.upper() != "PART" and self.profile.supports('paperFullCut'):
```

Comment:
Moreover the full cut should also happen in the case that `mode` is not supplied.

For the case that `paperFullCut` is not available but `paperPartCut` is, defaulting to a partial cut would be good.